result_pipe.py

```diff
@@ -0,0 +1,106 @@
+from livekit import rtc 
+from typing import Literal, Optional, Dict, Any, Callable
+from dataclasses import dataclass, field
+import uuid
+from datetime import datetime
+import json
+import time
+
+
+
+
+CLASSIFIER_RESULT_TOPIC = "classifier-result-topic"
+DETECTOR_RESULT_TOPIC = "detector-result-topic"
+TRACKER_RESULT_TOPIC = "tracker-result-topic"
+TOPICS_LIST = [CLASSIFIER_RESULT_TOPIC, DETECTOR_RESULT_TOPIC, TRACKER_RESULT_TOPIC]
+TOPICS_TYPE = Literal[CLASSIFIER_RESULT_TOPIC, DETECTOR_RESULT_TOPIC, TRACKER_RESULT_TOPIC]
+
+
+async def connect_to_room(url, token):
+    room = rtc.Room()
+    await room.connect(url, token)
+    return room
+
+
+def get_unique_id():
+    return str(uuid.uuid4())
+
+
+# an implementation of receiving results via data channel
+# following LiveKit's ChatManager interface
+
+@dataclass
+class ModelResult:
+    message: Optional[str] = None
+    id: str = field(default_factory=get_unique_id)
+    timestamp: datetime = field(default_factory=datetime.now)
+    deleted: bool = field(default=False)
+
+    # These fields are not part of the wire protocol. They are here to provide
+    # context for the application.
+    participant: Optional[rtc.room.Participant] = None
+    is_local: bool = field(default=False)
+
+    @classmethod
+    def from_jsondict(cls, d: Dict[str, Any]) -> "ModelResult":
+        # older version of the protocol didn't contain a message ID, so we'll create one
+        id = d.get("id") or get_unique_id()
```

Comment:
to clarify: we create a new uuid for _every_ message, not just for each stream?

Reply:
I believe that is what's happening yes